Core/GDCore/Events/Builtin/ForEachStructureEvent.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#ifndef FOREACHSTRUCTUREEVENT_H
+#define FOREACHSTRUCTUREEVENT_H
+#include "GDCore/Events/Event.h"
+#include "GDCore/Events/EventsList.h"
+namespace gd {
+class Instruction;
+class Project;
+class Layout;
+}  // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Event repeated for each every child of a structure variable.
+ */
+class GD_CORE_API ForEachStructureEvent : public gd::BaseEvent {
+ public:
+  ForEachStructureEvent();
+  virtual ~ForEachStructureEvent(){};
+  virtual gd::ForEachStructureEvent* Clone() const {
+    return new ForEachStructureEvent(*this);
+  }
+
+  virtual bool IsExecutable() const { return true; }
+
+  virtual bool CanHaveSubEvents() const { return true; }
+  virtual const gd::EventsList& GetSubEvents() const { return events; };
+  virtual gd::EventsList& GetSubEvents() { return events; };
+
+  const gd::InstructionsList& GetConditions() const { return conditions; };
+  gd::InstructionsList& GetConditions() { return conditions; };
+
+  const gd::InstructionsList& GetActions() const { return actions; };
+  gd::InstructionsList& GetActions() { return actions; };
+
+  /**
+   * \brief Get the structure attached to the event.
+   *
+   * The structure is the structure variable that will be iterated on.
+   */
+  const gd::String& GetStructure() const { return structure; };
+
+  /**
+   * \brief Set the structure attached to the event.
+   *
+   * The structure is the structure variable that will be iterated on.
+   */
+  void SetStructure(gd::String newStructure) { structure = newStructure; };
+
+  /**
+   * \brief Get the variable attached to the event.
+   *
+   * The variable is the variable that will contain the value of the linked
+   * child of the iteration.
+   */
+  const gd::String& GetVariable() const { return variable; };
+
+  /**
+   * \brief Set the variable attached to the event.
+   *
+   * The variable is the variable that will contain the value of the linked
+   * child of the iteration.
+   */
+  void SetVariable(gd::String newVariable) { variable = newVariable; };
```

Comment:
Similarly, let's call it the Get/SetIteratorVariableName

"Iterator Variable" because an "iterator" is what is being called for each iteration on the iterable. This iterator has a parameter which is the child, which we call the "Iterator Variable".
"Name" because it's a name that we're storing here.